test/conformance/websocket_test.go

```diff
@@ -0,0 +1,147 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	connectRetryInterval = 1 * time.Second
+	connectTimeout       = 6 * time.Minute
+)
+
+// connect attempts to establish websocket connection with the Service.
+// It will retry until reaching `connectTimeout` duration.
+func connect(logger *logging.BaseLogger, ingressIP string, domain string) (*websocket.Conn, error) {
+	u := url.URL{Scheme: "ws", Host: ingressIP, Path: "/"}
+	var conn *websocket.Conn
+	waitErr := wait.PollImmediate(connectRetryInterval, connectTimeout, func() (bool, error) {
+		logger.Infof("Connecting using websocket: url=%s, host=%s", u.String(), domain)
+		c, resp, err := websocket.DefaultDialer.Dial(u.String(), map[string][]string{"Host": []string{domain}})
+		if err == nil {
+			logger.Info("Connection established.")
+			conn = c
+			return true, nil
+		}
+		body := new(bytes.Buffer)
```

Comment:
Don't you need to close `resp.Body`?
You can also just go with `&bytes.Buffer{}`, instead of `new` call.